### PR TITLE
Closes #574 Adding az_menu block_content type.

### LINKED ISSUE #574: Create Menu block block type
Match Quickstart 1 implementation. Should be added in Content > Blocks.
- Have menu selection
- Allow selection of start level
- Allow selection of expand children
- Should be embeddable in paragraphs
- Should be organizable in Structure > Blocks

QS1 version 
image
image

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Go to /pages/menu-blocks
See demo content
Navigate through the menu to see behavior

Go to admin/structure/block/block-content
See demo blocks

Go to /admin/structure/block/block-content/types
See "Menu Block"

Go to /admin/structure/block/block-content/manage/az_custom_menu_block/fields
See fields

Go to /admin/structure/block/block-content/manage/az_custom_menu_block/form-display
See form display settings

Go to /admin/structure/block/block-content/manage/az_block/display 
See view mode 

Go to /node/add
Create new Flexible page
Add text area
Click Block embed button
Find Vertical Pills 1
Embed

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/templates/navigation/menu--block-content--az-menu--full--field-menu.html.twig [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{#
```
Comment (joeparsons) [inline themes/custom/az_barrio/templates/navigation/menu--block-content--az-menu--full--field-menu.html.twig:1]: Is this file still needed?  If so, does it need to be renamed (since the filename still references `field-menu`)?  Maybe `menu--block-content--az-block--full--field-az-block.html.twig`?

**Comment by danahertzberg** [comment]
## To do

- [ ] Change machine name to `az_custom_menu_block`
- [ ] Change module label to `Quickstart Block Types - Menu Block`
- [ ] Change block label to `Menu Block`
- [ ] Change block description to `An embeddable menu block.`
- [ ] Change field_az_block label to `Menu`
- [ ] Restrict field_az_block selections to `Main navigation (menu_block:main)` and `Utility Links (menu_block:az-utility-links)`
- [ ] Investigate removal of sidebar twig template file

## To test

- [ ] Document functionality of all options
- [ ] Determine good demo content

**Comment by danahertzberg** [comment]
Example use case: https://eller.arizona.edu/programs/mba/full-time/program

**Comment by trackleft** [comment]
In an effort to simplify selection of the template that the menu that is referenced within the block field, I went off and experimented with adding a view mode for this block type called az_vertical_pills, but I think it may add more confusion than it is worth:

- Need to update the embed button to allow selecting view modes other than the default.
- Embed button affects all block embeds
- Don't want to allow non-menu blocks to use the az_verticla_pills view mode (would be confusing)
- field_block doesn't pass the block view mode to the menu render step anyway.

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name: 'Quickstart Block Types - Menu'
+type: module
+description: 'Adds a block for placing blocks anywhere.'
```
Comment (camikazegreen) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml:3]:
```suggestion
description: 'Adds a block for placing menu blocks anywhere.'
```

File: modules/custom/az_demo/az_demo.info.yml [hunk 3]
```diff
@@ -4,6 +4,7 @@ description: 'WARNING:Do not enable this on live/production sites, it will alter
 core_version_requirement: ^9
 package: 'The University of Arizona - Experimental'
 dependencies:
+  - az_custom_menu_block
```
Comment (camikazegreen) [inline modules/custom/az_demo/az_demo.info.yml:7]:
```suggestion
  - az_menu_block
```

Comment (joeparsons) [reply modules/custom/az_demo/az_demo.info.yml:7]: `az_custom_menu_block` is the machine name we agreed on during the Wednesday workshop (to distinguish these from `menu_block` module blocks).

Comment (joeparsons) [reply modules/custom/az_demo/az_demo.info.yml:7]: Oh, but I see this is referencing the module name.  Never mind.

**Comment by trackleft** [comment]
I believe the background color of the block is transparent BTW.

File: modules/custom/az_block_types/az_custom_menu_block/config/install/block_content.type.az_custom_menu_block.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+langcode: en
+status: true
+dependencies:
+  enforced:
+    module:
+      - az_custom_menu_block
+id: az_custom_menu_block
+label: 'Menu Block'
+revision: 0
+description: 'An embeddable menu block with navigation view mode styles.'
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/config/install/block_content.type.az_custom_menu_block.yml:10]:
```suggestion
description: 'An embeddable content block with a block field that accepts menu blocks.'
```

File: themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+{% macro menu_links(items, attributes, menu_level) %}
+  {%
+    set nav_pills_ul_classes = [
+      'nav',
+      'flex-column',
+      'nav-pills',
+      'list-group',
+    ]
+  %}
+  {% import _self as menus %}
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig:37]:
```suggestion
<div class="bg-white">
  {% import _self as menus %}
```

File: themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+ *   - url: The menu link url, instance of \Drupal\Core\Url
+ *   - localized_options: Menu link localized options.
+ *   - is_expanded: TRUE if the link has visible children within the current
+ *     menu tree.
+ *   - is_collapsed: TRUE if the link has children within the current menu tree
+ *     that are not currently visible.
+ *   - in_active_trail: TRUE if the link is in the active trail.
+ */
+#}
+{% import _self as menus %}
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig:21]:
```suggestion
```

File: themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig (hunk 5 above)
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig:37]:
```suggestion
```

File: themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        {{ link(item.title, item.url, {'class': link_classes}) }}
+        {% if item.below %}
+          {{ menus.menu_links(item.below, attributes, menu_level + 1) }}
+        {% endif %}
+      </li>
+    {% endfor %}
+    </ul>
+  {% endif %}
+{% endmacro %}
+</div>
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--vertical-pills.html.twig:66]:
```suggestion
</div>

```

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+name: 'Quickstart Block Types - Menu'
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml:1]:
```suggestion
name: 'Quickstart Block Types - Menu Block'
```

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+type: module
+description: 'Adds an embeddable content block for placing menu blocks anywhere.'
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml:3]:
```suggestion
description: 'An embeddable content block with a block field that accepts menu blocks.'
```

File: modules/custom/az_block_types/az_custom_menu_block/config/install/block_content.type.az_custom_menu_block.yml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+langcode: en
+status: true
+dependencies:
+  enforced:
+    module:
+      - az_custom_menu_block
+id: az_custom_menu_block
+label: 'Menu Block'
+revision: 0
+description: 'An embeddable content block with a block field that accepts menu blocks.'
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/config/install/block_content.type.az_custom_menu_block.yml:10]:
```suggestion
description: 'Adds an embeddable content block for placing menu blocks anywhere.'
```

**Comment by danahertzberg** [comment]
- [ ] Needs permissions added for content admins (match menu permissions)

**Comment by danahertzberg** [comment]
Workshop topic: theme file consolidation

**Comment by trackleft** [comment]
Problem 1 (view mode selected templates). 
Block_field does not pass any indication of its paternity over the embedded menu block and subsequently menu_ _block's paternity over the menu it is referencing which makes switching the menu template based on the block containing the block_field's referenced block difficult.

Problem 2 (Set a default template suggestion for the menu_block's referenced menu)
hook_form_alter could possibly target the ajax loaded form for the menu_block once a menu is selected, and it possibly will be able to target only menu_blocks referenced in block fields, but I haven't been able to do it.

File: composer.json [hunk 11]
```diff
@@ -102,6 +103,7 @@
         "drupal/views_bootstrap": "5.4.0",
         "drupal/viewsreference": "2.0.0-beta3",
         "drupal/webform": "6.1.3",
+        "drupal/webprofiler": "^9.0",
```
Comment (trackleft) [inline composer.json:106]:
```suggestion
```

File: modules/custom/az_demo/data/az_demo_page_node.json [hunk 12]
```diff
@@ -129,6 +129,36 @@
...
+      "path" : "/pages/menu-blocks",
+      "exclude_title" : false,
+      "az_page_category"  : "",
+      "az_main_content"  : "79|80|81",
+      "az_media_image"    : "",
+      "az_summary" : ""
+    },
+    {
+      "id":15,
+      "title"  : "Menu Blocks",
```
Comment (bberndt-uaz) [inline modules/custom/az_demo/data/az_demo_page_node.json:145]:
```suggestion
      "title"  : "Menu Blocks - First Child",
```

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+   *   The current route match.
+   */
+  public function __construct(RouteMatchInterface $current_route_match) {
+    $this->currentRouteMatch = $current_route_match;
+  }
+
+  /**
+   * Alters the block library modal.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
```
Comment (bberndt-uaz) [inline modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:38]: I may be way off on this, but to resolve the PHPStan errors, does this need to reference GetResponseForControllerResultEvent?
- [getControllerResult()](https://api.drupal.org/api/drupal/vendor%21symfony%21http-kernel%21Event%21GetResponseForControllerResultEvent.php/function/GetResponseForControllerResultEvent%3A%3AgetControllerResult/9.3.x)
- [setControllerResult()](https://api.drupal.org/api/drupal/vendor%21symfony%21http-kernel%21Event%21GetResponseForControllerResultEvent.php/function/GetResponseForControllerResultEvent%3A%3AsetControllerResult/9.3.x)

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+      case 'block.admin_library':
+      case 'context.reaction.blocks.library':
+        // Grab the render array result before it is rendered by the
+        // main_content_view_subscriber.
+        $result = $event->getControllerResult();
+        foreach ($result['blocks']['#rows'] as $key => $row) {
+          // Remove rows for any block provided by the system_menu_block plugin.
+          $routeParameters = $row['operations']['data']['#links']['add']['url']->getRouteParameters();
+          $plugin_id = !empty($routeParameters['plugin_id']) ? $routeParameters['plugin_id'] : $routeParameters['block_id'];
+          if (strpos($plugin_id, 'menu_block:') === 0) {
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:52]:
```suggestion
          if (strpos($plugin_id, 'az_menu_block:') === 0) {
```

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php (hunk 13 above)
Comment (trackleft) [reply modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:38]: Let's try it.

Comment (trackleft) [reply modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:38]: Actually, that is deprecated https://api.drupal.org/api/drupal/vendor%21symfony%21http-kernel%21Event%21GetResponseForControllerResultEvent.php/9.3.x

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+    $this->currentRouteMatch = $current_route_match;
+  }
+
+  /**
+   * Alters the block library modal.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
+   *   The event to process.
+   */
+  public function onView(RequestEvent $event) {
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:41]:
```suggestion
  public function onKernelView(ViewEvent $event) {
```
Possibly

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php [hunk 16]
```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Drupal\az_custom_menu_block\EventSubscriber;
+
+use Drupal\Core\Routing\RouteMatchInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:7]:
```suggestion
use Symfony\Component\HttpKernel\Event\ViewEvent;
```

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php (hunk 13 above)
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:38]:
```suggestion
   * @param \Symfony\Component\HttpKernel\Event\ViewEvent;
 $event
```

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  public function __construct(RouteMatchInterface $current_route_match) {
+    $this->currentRouteMatch = $current_route_match;
+  }
+
+  /**
+   * Alters the block library modal.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\ViewEvent;
+ $event
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:39]:
```suggestion
   * @param \Symfony\Component\HttpKernel\Event\ViewEvent $event
```

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.services.yml [hunk 18]
```diff
@@ -0,0 +1,6 @@
+services:
+  az_custom_menu_block.az_kernel_view_subscriber:
```
Comment (joeparsons) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.services.yml:2]: Nitpick: I don't think the `az_` prefix is needed since the service is already prefixed with the module name.
```suggestion
  az_custom_menu_block.kernel_view_subscriber:
```

File: modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+    $this->currentRouteMatch = $current_route_match;
+  }
+
+  /**
+   * Alters the block library modal.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\ViewEvent $event
+   *   The event to process.
+   */
+  public function onKernelView(ViewEvent $event) {
```
Comment (joeparsons) [inline modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php:41]: Either this line needs to change or line 67 does.
```suggestion
  public function onView(ViewEvent $event) {
```

**Comment by trackleft** [comment]
Seems to work now

File: modules/custom/az_block_types/az_custom_menu_block/src/Plugin/Block/AzCustomMenuBlock.php [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+ * )
+ */
+class AzCustomMenuBlock extends MenuBlock {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    $defaults = parent::defaultConfiguration();
+    $defaults['suggestion'] = strtr('vertical_pills', '-', '_');
```
Comment (trackleft) [inline modules/custom/az_block_types/az_custom_menu_block/src/Plugin/Block/AzCustomMenuBlock.php:27]: Old default https://git.drupalcode.org/project/menu_block/-/blob/8.x-1.x/src/Plugin/Block/MenuBlock.php#L366

**Comment by trackleft** [comment]
Introducing a breaking change for anyone targeting anything like
```
  .sidebar ul.menu
  .sidebar ul.menu li 
  .sidebar ul.menu li > a,
  .sidebar ul.menu li > a:link,
  .sidebar ul.menu li > a:visited 
  .region-sidebar-first .menu > li > a:hover
  .region-sidebar-first .menu > li > a:focus
  .region-sidebar-first .menu > li.active a
  .region-sidebar-first .menu > li a.active 
  .region-sidebar-second .menu > li > a:hover
  .region-sidebar-second .menu > li > a:focus,
  .region-sidebar-second .menu > li.active a
  .region-sidebar-second .menu > li a.active
  ```

File: themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig [hunk 21]
```diff
@@ -18,21 +18,25 @@
...
+      'list-group',
+    ]
+  %}
   {% if items %}
-    {% if menu_level == 0 %}
-      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-pills').addClass('list-group') }}>
-    {% else %}
-      <ul class="menu">
-    {% endif %}
+    <ul{{ attributes.addClass(nav_pills_ul_classes) }}>
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig:39]:
```suggestion
    <ul{{ attributes.addClass(nav_pills_ul_classes).addClass('menu') }}>
```
For backwards compatibility

**Comment by trackleft** [comment]
Blocked by https://github.com/az-digital/arizona-bootstrap/pull/555

**Comment by trackleft** [comment]
How do we feel about removing the `menu` class from the second level (and below) `ul` element within the sidebar menu template? https://github.com/az-digital/az_quickstart/pull/1887/files#diff-2768534006e21c219df9520d6cb4b5706eaf7ee64a2a18e45024bd5c2e0c5d28L31-L35

**Comment by bberndt-uaz** [comment]
> How do we feel about removing the `menu` class from the second level (and below) `ul` element within the sidebar menu template? https://github.com/az-digital/az_quickstart/pull/1887/files#diff-2768534006e21c219df9520d6cb4b5706eaf7ee64a2a18e45024bd5c2e0c5d28L31-L35

Just so I understand:
- We could remove the `menu` class since this PR removes the only references to it (from style.css) and so the class is no longer needed. Is that right?
- Are there any other risks to removing it besides the possibility that someone has a customization targeting that selector?

**Comment by trackleft** [comment]
> * Are there any other risks to removing it besides the possibility that someone has a customization targeting that selector?

Yes the risk is that someone has made a customization that targets that class.

**Comment by trackleft** [comment]
I am testing your suggestion locally @tadean  Thanks!

File: themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig [hunk 22]
```diff
@@ -19,20 +19,23 @@
...
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig:38]:
```suggestion
    <ul{{ attributes.addClass(nav_pills_ul_classes).addClass(menu}}>
```

File: themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig [hunk 23]
```diff
@@ -19,20 +19,23 @@
...
+      'nav-pills',
+      'list-group',
+    ]
+  %}
   {% if items %}
-    {% if menu_level == 0 %}
-      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-pills').addClass('list-group') }}>
-    {% else %}
-      <ul class="menu">
-    {% endif %}
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig:35]:
```
{% if menu_level == 0 %}
      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-pills').addClass('list-group') }}>
    {% else %}
      <ul{{ attributes.addClass(nav_pills_ul_classes).addClass(menu}}>
{% endif %}
```

File: themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig [hunk 24]
```diff
@@ -19,20 +19,23 @@
...
 {{ menus.menu_links(items, attributes, 0) }}
+
 {% macro menu_links(items, attributes, menu_level) %}
-  {% import _self as menus %}
+  {%
+    set nav_pills_ul_classes = [
+      'nav',
+      'flex-column',
+      'nav-pills',
+      'list-group',
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig:34]:
```suggestion
      'bg-white',
```

File: themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig [hunk 25]
```diff
@@ -19,19 +19,26 @@
...
+    ]
+  %}
   {% if items %}
     {% if menu_level == 0 %}
-      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-pills').addClass('list-group') }}>
-    {% else %}
-      <ul class="menu">
+          <ul{{ attributes.addClass(nav_pills_ul_classes)}}>
+        {% else %}
+          <ul{{ attributes.addClass(nav_pills_ul_classes).addClass(menu)}}>
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig:41]:
```suggestion
      <ul{{ attributes.addClass(nav_pills_ul_classes)}}>
    {% else %}
      <ul{{ attributes.addClass(nav_pills_ul_classes).addClass(menu)}}>
```

File: modules/custom/az_demo/data/az_demo_block_content_block.json [hunk 26]
```diff
@@ -0,0 +1,80 @@
+{
+  "blocks": [
+    {
+      "id":1,
+      "info" : "Vertical Pills 1",
+      "uuid" : "8d5d7e34-7d98-489d-8415-fc0c72a6568f",
+      "field_az_block": [
+        {
+          "plugin_id": "az_menu_block:main",
```
Comment (trackleft) [inline modules/custom/az_demo/data/az_demo_block_content_block.json:9]:
```suggestion
          "plugin_id": "menu_block:main",
```

File: modules/custom/az_demo/data/az_demo_block_content_block.json [hunk 27]
```diff
@@ -0,0 +1,80 @@
...
+        }
+      ]
+    },
+    {
+      "id":2,
+      "info" : "Vertical Pills 2",
+      "uuid" : "3a7fffab-e13b-429e-9405-f2bb89c83d63",
+      "field_az_block": [
+        {
+          "plugin_id": "az_menu_block:main",
```
Comment (trackleft) [inline modules/custom/az_demo/data/az_demo_block_content_block.json:34]:
```suggestion
          "plugin_id": "menu_block:main",
```

File: modules/custom/az_demo/data/az_demo_block_content_block.json [hunk 28]
```diff
@@ -0,0 +1,80 @@
...
+      ]
+    },
+    {
+      "id":3,
+      "info" : "Vertical Pills 3",
+      "uuid" : "462e3902-55a4-4c2a-b7d5-b5814e394b9c",
+      "menu_parent_link_id": "26",
+      "field_az_block": [
+        {
+          "plugin_id": "az_menu_block:main",
```
Comment (trackleft) [inline modules/custom/az_demo/data/az_demo_block_content_block.json:60]:
```suggestion
          "plugin_id": "menu_block:main",
```

File: themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig [hunk 29]
```diff
@@ -19,19 +19,25 @@
...
+      'bg-white',
+    ]
+  %}
   {% if items %}
     {% if menu_level == 0 %}
-      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-pills').addClass('list-group') }}>
+      <ul{{ attributes.addClass(nav_pills_ul_classes)}}>
     {% else %}
-      <ul class="menu">
+      <ul{{ attributes.addClass(nav_pills_ul_classes).addClass(menu)}}>
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig:40]: I don't think this is working

Comment (trackleft) [inline themes/custom/az_barrio/templates/navigation/menu--sidebar.html.twig:40]:
```suggestion
      <ul{{ attributes.addClass(nav_pills_ul_classes).addClass('menu') }}>
```

**Comment by trackleft** [comment]
I've removed the breaking change checkbox FYI

File: modules/custom/az_demo/migrations/az_demo_block_content_menu.yml [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+      method: process
+    - plugin: concat
+      source:
+       - constants/MENU_ITEM_PREFIX
+       - '@pseudo_parent_uuid'
+      delimiter: ":"
+destination:
+  plugin: entity:block_content
+  bundle: az_custom_menu_block
+
```
Comment (joeparsons) [inline modules/custom/az_demo/migrations/az_demo_block_content_menu.yml:77]: Annoying nitpick (extra whitespace):
```suggestion
```

**Comment by trackleft** [comment]
I'll hold off on committing code suggestions until az-digital wednesday.

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml [hunk 31]
```diff
@@ -0,0 +1,9 @@
+name: 'Quickstart Block Types - Menu Block'
```
Comment (bberndt-uaz) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.info.yml:1]: Change to "Quickstart Block Types - Custom Menu Block"?

File: modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module [hunk 32]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Due to function name we only run on block_content_az_custom_menu_block_form.
+ * This is the creation form for az_custom_menu_block content.
+ */
+function az_custom_menu_block_form_block_content_az_custom_menu_block_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+
+  // Default suggesition value.
```
Comment (bberndt-uaz) [inline modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module:18]:
```suggestion
  // Default suggestion value.
```

File: modules/custom/az_block_types/az_custom_menu_block/config/install/block_content.type.az_custom_menu_block.yml [hunk 33]
```diff
@@ -0,0 +1,10 @@
+langcode: en
+status: true
+dependencies:
+  enforced:
+    module:
+      - az_custom_menu_block
+id: az_custom_menu_block
+label: 'Menu Block'
```
Comment (bberndt-uaz) [inline modules/custom/az_block_types/az_custom_menu_block/config/install/block_content.type.az_custom_menu_block.yml:8]: Change to "Custom Menu Block"?

**Review by joeparsons — DISMISSED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Comment by danahertzberg** [comment]
Need to create follow up issue for applying matching title styles on this menu to match sidebar menu title styles

**Review by tadean — APPROVED** [review]

**Comment by joeparsons** [comment]
Pre-release testing:
- Make sure that active and visited link styles are still correct in sidebar menus
